payment/src/main/java/org/killbill/billing/payment/dispatcher/CallableWithRequestData.java

```diff
@@ -72,7 +72,8 @@ public T call() throws Exception {
             UUIDs.setRandom(null);
             ThreadContext.unbindSecurityManager();
             ThreadContext.unbindSubject();
-            MDC.clear();
+            // Reset the MDC as before the call, in case the plugin fiddle'd with it
+            MDC.setContextMap(mdcContextMap);
```

Comment:
This 'plugin' thread is going back to the pool - don;t we want to clear the state completely - also how do we know this was the previous state?

Reply:
Absolutely - fixed.